pkg/domain/workflow.go

```diff
@@ -7,7 +7,24 @@ import (
 	"github.com/fuseml/fuseml-core/gen/workflow"
 )
 
-// WorkflowStore is an inteface to workflow stores
+const (
+	// ErrWorkflowExists describes the error message returned when trying to create a workflow with that already exists.
+	ErrWorkflowExists = WorkflowErr("workflow already exists")
+	// ErrWorkflowNotFound describes the error message returned when trying to get a workflow that does not exist.
+	ErrWorkflowNotFound = WorkflowErr("could not find a workflow with the specified name")
+	// ErrWorkflowNotAssignedToCodeset describes the error message returned when trying to unassign a workflow from a codeset
+	// but it is not assigned to the codeset.
+	ErrWorkflowNotAssignedToCodeset = WorkflowErr("workflow not assigned to codeset")
+)
```

Comment:
This is a good idea and I'm gonna use it for my PR too. Just one question - what about using `domain/errors.go` instead and gather all errors there instead of making them specific for each artifact?

Reply:
Do you mean having a generic type for errors instead of `WorkflowErr`? I think it makes sense as there is nothing particular to `WorkflowErr`.

Reply:
That, and also that all errors would be stored in one file